### ISSUE TITLE
State Store capabilities matrix could be more useful

### ISSUE THREAD

**Issue by olitomlinson** [dapr/docs#2739]
Currently the state store capability matrix only shows a simple checkbox, which I and I imagine most people interpret as being "Supported" / "Not Supported"

<img width="500" alt="image" src="https://user-images.githubusercontent.com/4224880/185749371-c53645d8-c5ea-4cf1-b523-4dbe29be4fad.png">

However I think, this could be expanded a little to include some other states.

For example, when I see a capability as "Not Supported" I never really know which of the two statements are true

1. "This capability is not technically possible/feasible due to constraints or limitations"
2. "It is actually possible, but it's just not implemented"

I think differentiating between the two is important because

1. If I know something is technically not possible, then I'm more than likely going to rule-out that store and chose an alternate.
2. If I know it is technically possible, but just not implemented, then I may chose to bring attention to this and attempt to get it prioritised for delivery (hell, even attempt to contribute some code myself, which is terrifying for everyone involved I'm sure!)

Another consideration is that **things change over time**

Something may have been technically impossible a year or two back, but as we all know, things move forwards, and maybe that capability that we believe is not technically possible is now quite possible today?! Therefore, how do we track when capabilities were last reviewed for correctness? Also it would be nice to review someones rationale on why they believe something isn't capable.

---

So in summary

- Is the capability implemented or not?
- If not, why not?
- If it's not technically feasible, when was the last time it was reviewed?
- and, what rationale was given for it not being feasible?

I wouldn't expect that the matrix wouldn't contain all this information as that would be a real info-overload, however I could imagine new states being introduced (as opposed to a boolean checkbox) and clickable links which reference individual long-lived GH Issues which track each implementation and the capabilities supported / not supported.

I hypothesise that if this information becomes more accessible, it might bring greater confidence and inspire creativity through discourse among the community to achieve even greater outcomes.

**Comment by yaron2** [comment]
cc @msfussell

**Comment by hhunter-ms** [comment]
@msfussell friendly bump - what do you think?
